### DISCUSSION TITLE
Next.js App API route returning 404 even though route is present

### DISCUSSION THREAD

**Discussion by mikr13** [vercel/next.js#74213] [Help]
### Summary

I have a `POST:/api/github` route in my next.js app.
Path to file: `src/app/api/github/route.ts`

When I try calling the API from postman, it returns 404 error, when I open up browser to just check in case, that also returns 404 error. I also don't have any conflicting paths in app router

PFA the screenshots for more details:

Directory ss:
<img width="351" alt="image" src="https://github.com/user-attachments/assets/4ca6a852-d4dd-4892-86b9-197b386c8c89" />

Ngrok API Response ss:
<img width="922" alt="image" src="https://github.com/user-attachments/assets/799cfdab-3eb8-480f-8627-c9363ead8fef" />

Next.js logs:
<img width="267" alt="image" src="https://github.com/user-attachments/assets/e5113af7-81be-4434-bf5d-7e5daca3644c" />

Code SS:
<img width="490" alt="image" src="https://github.com/user-attachments/assets/5d67373a-956a-4239-927e-56a443091410" />

Please help. Thanks!

### Additional information

    "next": "15.1.2",
    "react": "^19.0.0",

**Comment by eric-mxrtin** [comment]
I had a similar issue, try a full server restart and clearing your .next build folder.

Also, could be an issue with your Nextjs version?

**Reply by mikr13** [reply]
I did that, I deleted the .next & node_modules folder & tried. Still nothing. Let me see about downgrading to 14

**Reply by mikr13** [reply]
Downgrading to 14 didn't help either. Still getting POST 404 error

**Reply by mikr13** [reply]
I noticed something, so when I have some logic inside my route handler, it goes 404, but when I do something like:

```ts
import type { NextApiRequest } from 'next';

export async function POST(
  req: NextApiRequest,
) {
  return Response.json({ message: 'Hello from Next.js!' })
}
```
Then it works fine. Not sure what this behaviour is though.

**Comment by kuldip-dev** [comment]
@mikr13 Try below test code hope this will work.
```

// app/api/post/route.js

import { NextResponse } from 'next/server';

export async function POST(request) {
  try {
    // Parse the request body
    const body = await request.json();

    // Validate required fields
    if (!body.name || !body.email) {
      return NextResponse.json({ error: 'Name and email are required' }, { status: 400 });
    }

    // Example data processing
    const processedData = {
      id: Math.random().toString(36).substr(2, 9),
      name: body.name,
      email: body.email,
      timestamp: new Date().toISOString(),
      // Add any additional fields from body
      ...body
    };

    // Here you would typically save to a database
    // For this example, we'll just return the processed data
    return NextResponse.json(
      {
        message: 'Data received successfully',
        data: processedData
      },
      { status: 201 }
    );
  } catch (error) {
    console.error('POST request error:', error);
    return NextResponse.json({ error: 'Internal Server Error' }, { status: 500 });
  }
}
```
You can test this API using:

1. cURL:
```
curl -X POST http://localhost:3000/api/post \
-H "Content-Type: application/json" \
-d '{"name": "John Doe", "email": "john@example.com"}'
```
2. Thunder Client or Postman:
```
URL: http://localhost:3000/api/post
Method: POST
Headers: 
  Content-Type: application/json
Body:
{
  "name": "John Doe",
  "email": "john@example.com"
}
```
